src/core_plugins/kibana/public/management/sections/indices/index.js

```diff
@@ -133,13 +130,11 @@ uiModules.get('apps/management')
           updateIndexPatternList($scope, indexPatternCreationOptions, $scope.defaultIndex, $scope.indexPatternList);
         };
 
-        $scope.$watch('defaultIndex', renderList);
-
-        renderList();
-
-        $scope.$on('$destroy', () => {
-          destroyIndexPatternList();
-        });
+        $scope.$on('$destroy', () => destroyIndexPatternList());
+        $scope.editingId = $route.current.params.indexPatternId;
+        $scope.$watch('defaultIndex', () => renderList());
+        config.bindToScope($scope, 'defaultIndex');
+        $scope.$apply();
```

Comment:
Out of curiosity, what motivated these changes? And why call `$scope.$apply()`? I think it's generally considered a bad practice because it can cause a [`$digest already in progress` error](https://stackoverflow.com/questions/12729122/angularjs-prevent-error-digest-already-in-progress-when-calling-scope-apply). The safer way would be to wrap some code inside of `$evalAsync`.

Reply:
this is a change on top of a previous change that converted the sidebar to react and I ran into some issues with rendering the react component. I agree `$scope.$apply()` should be avoided, will take a closer look at this next pass with my work for scripted fields